src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticToken.java

```diff
@@ -254,6 +260,27 @@ LSPSemanticTokenPsiElement getElement() {
         return element;
     }
 
+    /**
+     * Stores the last requested offset for which a file-level semantic token was returned as a thread local.
+     *
+     * @param offset the last requested offset
+     */
+    void setLastRequestedOffset(int offset) {
+        if (isFileLevel) {
```

Comment:
Again, only for file-level, so this wouldn't be populated for an offset that corresponds to a semantic token PSI element.